node/src/isolate/process.cpp

```diff
@@ -399,12 +400,17 @@ process_t::~process_t() {
     extern char** environ;
 #endif
 
-std::unique_ptr<api::handle_t>
-process_t::spawn(const std::string& path, const api::string_map_t& args, const api::string_map_t& environment) {
+std::unique_ptr<api::cancellation_t>
+process_t::spawn(const std::string& path, const api::string_map_t& args, const api::string_map_t& environment, api::spawn_handler_t handler) {
     std::array<int, 2> pipes;
 
     if(::pipe(pipes.data()) != 0) {
-        throw std::system_error(errno, std::system_category(), "unable to create an output pipe");
+        COCAINE_LOG_ERROR(m_log, "unable to create an output pipe: %d", errno);
```

Comment:
When it's time to format the message an `errno` can be changed for some other reasons than the initial one.